sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -347,6 +347,14 @@ querySpecification
        windows?)
     ;
 
+hint
+    : '/*+' mapJoinHint '*/'
+    ;
+
+mapJoinHint
+    : MAPJOIN '(' broadcastedTables+=tableIdentifier (',' broadcastedTables+=tableIdentifier)* ')'
```

Comment:
Yep. MAPJOIN optimization will be handled in Optimizer. This PR is just accepting the basic syntax.